core/src/test/java/org/apache/iceberg/io/InMemoryOutputFile.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.io;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Objects;
+import java.util.UUID;
+
+public class InMemoryOutputFile implements OutputFile {
+
+  private final String location;
```

Comment:
> I believe this is to match the behavior of `file.toString()`, which might be relied on elsewhere.

toString representation mustn't be relied on, unless its contract is defined by an interface (it isn't here)

location-returning toString wouldn't be useful, as this is just a UUID.

Reply:
That's fair. I'm pretty sure we removed any and all reliance on inappropriate usages of `toString` quite some time ago. Since it's not part of the contract, it's just added overhead then. 👍